pkg/storage/allocator.go

```diff
@@ -491,6 +511,33 @@ func (a Allocator) RebalanceTarget(
 	if target == nil {
 		return nil, ""
 	}
+	// We could make a simulation here to verify whether we'll remove the target we'll rebalance to.
```

Comment:
Do you think we could avoid this code duplication with what's in `replicate_queue.go` and instead create a shared function they can both use? If they're separate like this, it seems likely that they'll accidentally diverge in the future.

Reply:
Hmm, Sorry, I couldn't quite understand what do you mean here. I didn't make any change to `replicate_queue.go`.
Would you please point out the  code duplication with `replicate_queue.go`? Thank you very much.

Reply:
I was referring to the code for the `AllocatorRemove` case in `processOneChange`, but on looking at it again it doesn't look like there's much that could be de-duped. Sorry for the confusion!